plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/WriteMapping.java

```diff
@@ -70,6 +87,11 @@ public WriteFunction getWriteFunction()
         return writeFunction;
     }
 
+    public WriteMapping withEffectiveType(Type effectiveType)
```

Comment:
We cannot verify that, because until we create the table, we don't have the input to feed into the `toColumnMapping`.

however, this also uncovers a problem with my earlier suggestion (https://github.com/trinodb/trino/pull/13994#discussion_r973617031)
i suggested to change `toWriteMapping`, so that it can optionally return "effective type"
This is OK for callers like `io.trino.plugin.jdbc.DefaultJdbcMetadata#getNewTableEffectiveMetadata` and maybe for `io.trino.plugin.jdbc.BaseJdbcClient#getColumnDefinitionSql`.
But it's not OK for `JdbcPageSink::JdbcPageSink` -- the `JdbcPageSink` is too late to apply any "effective type".

Now as I am thinking about this, let's not put that into `toWriteMapping` method, but let's add a new method (that will be invoked from `DefaultJdbcMetadata#getNewTableEffectiveMetadata` for each column separately)